sparse/unit_test/Test_vector_fixtures.hpp

```diff
@@ -50,35 +56,40 @@ void compress_matrix(
   }
 
   // Allocate device CRS views
-  Kokkos::resize(row_map, nrows + 1);
+  Kokkos::resize(map, (CSC ? ncols : nrows) + 1);
   Kokkos::resize(entries, nnz);
   Kokkos::resize(values, nnz);
 
   // Create host mirror views for CRS
-  auto hrow_map = Kokkos::create_mirror_view(row_map);
+  auto hmap     = Kokkos::create_mirror_view(map);
   auto hentries = Kokkos::create_mirror_view(entries);
   auto hvalues  = Kokkos::create_mirror_view(values);
 
   // Compress into CRS (host views)
   size_type curr_nnz = 0;
-  for (size_type row_idx = 0; row_idx < nrows; ++row_idx) {
-    for (size_type col_idx = 0; col_idx < nrows; ++col_idx) {
-      if (fixture[row_idx][col_idx] != ZERO) {
-        hentries(curr_nnz) = col_idx;
-        hvalues(curr_nnz)  = fixture[row_idx][col_idx];
+  for (size_type outer_idx = 0; outer_idx < (CSC ? ncols : nrows);
+       ++outer_idx) {
+    for (size_type inner_idx = 0; inner_idx < (CSC ? nrows : ncols);
+         ++inner_idx) {
+      const auto val =
+          fixture[CSC ? inner_idx : outer_idx][CSC ? outer_idx : inner_idx];
+      if (val != ZERO) {
+        hentries(curr_nnz) = inner_idx;
+        hvalues(curr_nnz)  = val == KEEP_ZERO<scalar_t>() ? ZERO : val;
```

Comment:
@lucbv , there were a few cases in the sptrsv test where the CRS was being populated with zeroes in some of their values:

Example in some of the old code
```
-      // values
-      // first row (first supernode)
-      hUvalues(0) = FIVE;
-      hUvalues(1) = ONE;
-      hUvalues(2) = ONE;
-      hUvalues(3) = ZERO;
-      // second row (first supernode)
-      hUvalues(4) = ZERO;
-      hUvalues(5) = FIVE;
-      hUvalues(6) = ZERO;
-      hUvalues(7) = ONE;
-      // third row (second supernode)
-      hUvalues(8)  = FIVE;
-      hUvalues(9)  = ONE;
-      hUvalues(10) = ONE;
-      // fourth row (third supernode)
-      hUvalues(11) = FIVE;
-      hUvalues(12) = ONE;
-      // fifth row (fourth supernode)
-      hUvalues(13) = FIVE;
```

The fixture code normally skips zeroes when it's converting the 2d vector to CRS, so some sentinel value was needed so that it would know to keep the zero:

```
  static std::vector<std::vector<scalar_t>> get_5x5_ut_fixture() {
    const auto KZ                        = KEEP_ZERO<scalar_t>();
    std::vector<std::vector<scalar_t>> A = {{5.00, 1.00, 1.00, 0.00, KZ},
                                            {KZ, 5.00, KZ, 0.00, 1.00},
                                            {0.00, 0.00, 5.00, 1.00, 1.00},
                                            {0.00, 0.00, 0.00, 5.00, 1.00},
                                            {0.00, 0.00, 0.00, 0.00, 5.00}};
    return A;
  }
```

Reply:
okay